app/controllers/api/storage.php

```
@@ -245,6 +246,218 @@
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_FILE)
+    ->param('bucketId', null, new UID(), 'Storage bucket unique ID. You can create a new storage bucket using the Storage service [server integration](/docs/server/storage#createBucket).')
+    ->param('file', [], new File(), 'Binary file.', false)
+    ->param('read', null, new ArrayList(new Text(64)), 'An array of strings with read permissions. By default only the current user is granted with read permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->param('write', null, new ArrayList(new Text(64)), 'An array of strings with write permissions. By default only the current user is granted with write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->inject('request')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('user')
+    ->inject('audits')
+    ->inject('usage')
+    ->action(function ($bucketId, $file, $read, $write, $request, $response, $dbForInternal, $user, $audits, $usage) {
+        /** @var Utopia\Swoole\Request $request */
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Database\Document $user */
+        /** @var Appwrite\Event\Event $audits */
+        /** @var Appwrite\Event\Event $usage */
+
+        $bucket = $dbForInternal->getDocument('buckets',$bucketId);
+
+        if($bucket->isEmpty()) {
+            throw new Exception("Unable to find the bucket", 404);
+        }
+
+        $file = $request->getFiles('file');
+
+        /*
+         * Validators
+         */
+        // $fileType = new FileType($bucket->getAttribute('allowedFileExtensions', ['*']));
+        $fileSize = new FileSize($bucket->getAttribute('maximumFileSize', 0));
```

Should we throw an error here if bucket's `maximumFileSize` is bigger than `_APP_STORAGE_LIMIT`? Just in case `_APP_STORAGE_LIMIT` was changed after the bucket created, what do you say?

---

OK, let's add it, I think we can also consider it as a server error (500), WDYT?

---

Good enough!